site/docs/features/index.md

```diff
@@ -1,107 +1,327 @@
+  the actual data
+* Add [meanings to the changes](#commit-messages-and-more) to your data lake
+* [Always-consistent](#branches) view to all the data 
+* Sets of changes, like the whole [work of a distributed Spark job](#working-branches-for-analytics-jobs)
+  or [experiments of data engineers](#working-branches-for-humans), are
+  isolated in Nessie via *branches*. Failed jobs do not add additional harm to the data.
+* Known, fixed versions of all data can be [tagged](#tags)
+* Automatic removal of unused data files ([garbage collection](#garbage-collection))
+
+## Data Lake 101
+
+> *"A data lake is a system or repository of data stored in its natural/raw format,
+usually object blobs or files."* (cite from [Wikipedia](https://en.wikipedia.org/wiki/Data_lake))
+
+Data is stored in immutable data files[^1]. Each data file defines the schema of the
+data (i.e. names and types of the columns) and contains the data. A single,
+logical table (for example a `customers` or a `bank_account_transactions` table)
+consists of many data files.
+
+New data files can be added to the set of files for a particular table.
+Data files can also contain updates to and deletions of existing data.
+
+Since data lakes contain the history of all that data, it is possible to
+perform queries against the state of the data as it was at a point in time
+in the past.
+
+The amount of data held in data lakes is rather huge (GBs, TBs, PBs), and so
+is the number of tables and data files (100s of thousands, millions).
+
+Managing that amount of data and data files while keeping track of schema
+changes, for example adding or removing a column, changing a column's type,
+renaming a column in a table ("PDS", physical datasets) and views ("VDS",
+virtual datasets), is one of the things that Nessie tackles.
+
+Data in a data lake is usually consumed and written using tools like
+Apache [Hive](https://hive.apache.org)[^2] or Apache
+[Spark](https://spark.apache.org)[^2]. Your existing jobs can easily integrate
+Nessie without any production code changes, it's a simple configuration change.
+
+## git 101
+
+> *"Git is a free and open source distributed version control system designed to 
+handle everything from small to very large projects with speed and efficiency"*
+(cite from [git-scm.com](https://git-scm.com/))
+
+git maintains the history or all changes of a software project from the very
+first *commit* until the current state.
+
+git is used by humans, i.e. developers.
+
+Many of the concepts of git for source code are implemented by Nessie for all
+the data your data lake. It would be rather confusing to explain all git
+concepts here and then outline the differences in the next chapter.
+
+## Working with data in Nessie
+
+Each individual state in Nessie is defined by a *Nessie commit*.
+Each *commit* in Nessie, except the very first one, has references to its
+predecessors, the previous versions of the data.
```

Comment:
Is it even for a merge-operation?

Reply:
yep

Reply:
I think it would be very useful to keep the information about "what" has been merged.
Or, if we stick with the rather "rebase-ish" operation (i.e. apply the "work branch"'s commits onto the main branch), maybe another term would be useful, because the term "merge" has a well defined meaning in Git.
/cc @jacques-n WDYT?

Reply:
Changes some paragraphs to reflect this.